turbo/transactions/call.go

```diff
@@ -45,6 +48,24 @@ func DoCall(
 			return state, block.Header(), nil
 		}
 	*/
+	num := rpc.LatestBlockNumber
+	latestBlockNumber, _, _, err := rpchelper.GetBlockNumber(rpc.BlockNumberOrHash{BlockNumber: &num}, tx, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	pruneAmount, err := prune.Get(tx)
+	if err != nil {
+		return nil, err
+	}
+
+	latestPrunedBlockNumber := latestBlockNumber - pruneAmount.CallTraces.ToValue()
```

Comment:
I added a check so if latestFinishedBlockNumber < pruneAmount then we just return true otherwise we check if latestFinishedBlockNumber-pruneAmount >= requestedBlockNumber